pywren_ibm_cloud/wren.py

```diff
@@ -336,10 +336,10 @@ def monitor(self, futures=None, throw_except=True, return_when=ALL_COMPLETED,
                 self.clean()
 
         if download_results:
-            fs_dones = [f for f in ftrs if f.done]
+            fs_dones = [f for f in ftrs if f.done and not f._is_map_before_reduce]
             fs_notdones = [f for f in ftrs if not f.done]
         else:
-            fs_dones = [f for f in ftrs if f.ready]
+            fs_dones = [f for f in ftrs if f.ready and not f._is_map_before_reduce]
```

Comment:
@JosepSampe by this line, map futures will never produce results as I said before :)
but I think it is preferred to be implemented inside `result()` method of `future.py` as you suggested.

Reply:
@omerb01 Yep, the results are not returned to the user with this line, however the get_result() method in futures.py is called from wait.py, so results are downloaded. Map futures will be monitored since PyWren will download the statuses, so we will know in logs when maps finished, and if they produced error or not. Also we will be able to print plots with only statuses.